betterlockscreen

```diff
@@ -4,68 +4,34 @@
 # Github Profile : https://github.com/pavanjadhaw
 # Project Repository : https://github.com/pavanjadhaw/betterlockscreen
 
+lockargs=(-n)
+
 cmd_exists () {
     command -v "$1" >/dev/null
 }
 
 init_config () {
+    # Load defaults
+    source "./defaultsrc"
 
-    # default options
-    display_on=0
-    span_image=false
-    lock_timeout=300
-    fx_list=(dim blur dimblur pixel dimpixel color)
-    dim_level=40
-    blur_level=1
-    pixel_scale=10,1000
-    solid_color=333333
-    description=""
-    quiet=false
-    i3lockcolor_bin="i3lock-color"
-
-    if ! cmd_exists "$i3lockcolor_bin" && cmd_exists "i3lock"; then
-        i3lockcolor_bin="i3lock"
-    fi
-
-    # default theme
-    loginbox=00000066
-    loginshadow=00000000
-    locktext="Type password to unlock..."
-    font="sans-serif"
-    ringcolor=ffffffff
-    insidecolor=00000000
-    separatorcolor=00000000
-    ringvercolor=ffffffff
-    insidevercolor=00000000
-    ringwrongcolor=ffffffff
-    insidewrongcolor=d23c3dff
-    timecolor=ffffffff
-    time_format="%H:%M:%S"
-    greetercolor=ffffffff
-    layoutcolor=ffffffff
-    keyhlcolor=d23c3dff
-    bshlcolor=d23c3dff
-    verifcolor=ffffffff
-    wrongcolor=d23c3dff
-    modifcolor=d23c3dff
-    bgcolor=000000ff
-    wallpaper_cmd="feh --bg-fill"
-    time_format="%H:%M:%S"
-
-    # read user config
-    USER_CONF="${XDG_CONFIG_HOME:-$HOME/.config}/betterlockscreenrc"
+    CONF_DIR="${XDG_CONFIG_HOME:-$HOME/.config}"
+    USER_CONF="$CONF_DIR/betterlockscreenrc"
```

Comment:
We could also optionally check for `/etc/betterlockscreenrc`, as this would allow packagers/distros to provide at custom or distro-specific config.

Something like this:
```bash
    CONF_DIR="${XDG_CONFIG_HOME:-$HOME/.config}"
    USER_CONF="$CONF_DIR/betterlockscreenrc"
    SYS_CONF="/etc/betterlockscreenrc"
    if [ -e "$USER_CONF" ]; then
        # shellcheck source=/dev/null
        source "$USER_CONF"
    elif [ -e "$SYS_CONF" ]; then
        # shellcheck source=/dev/null
        source "$SYS_CONF"
    fi
```